From f9b20defe5f45cffa93a7264fd9c0b137a5ec9f2 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Sat, 11 Mar 2023 22:26:22 +0000
Subject: [PATCH] CMakeLists.txt: remove dependency on libprocps

:Release Notes:
Remove unused dependency on libprocps.

:Detailed Notes:
Fixes build with procps-4:
http://gecko.lge.com:8000/Errors/Details/551318

-- Checking for module 'libprocps'
--   No package 'libprocps' found

there might be many more changes needed to adapt to procps 4 API, see:
https://gitlab.com/procps-ng/procps/-/issues/239

procps upgrade was now merged in oe-core together with igt-gpu-tools adaptation in:
https://git.openembedded.org/openembedded-core/commit/?id=631eba02d64f1a42514e0ae4361bbecc5cce5fa7

but luckily it's not actually being used here other than this pkg-config
check, so just remove it.

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-10874] Prepare for procps 4 API

Change-Id: Ifd45a298f64afd4550fb3384926a582dcb1c6a50
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/sam/+/347392 CMakeLists.txt: remove dependency on libprocps]

 CMakeLists.txt | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9ffa77a..094e43d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,10 +66,6 @@ find_package(Boost REQUIRED COMPONENTS regex system filesystem)
 include_directories(${Boost_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${Boost_CFLAGS_OTHER})
 
-pkg_check_modules(PROCPS REQUIRED libprocps)
-include_directories(PROCPS_INCLUDE_DIRS)
-webos_add_compiler_flags(ALL ${PROCPS_CFLAGS})
-
 find_library(ICU NAMES icuuc)
 if(ICU STREQUAL "ICU-NOTFOUND")
    message(FATAL_ERROR "Failed to find ICU4C libraries. Please install.")
@@ -103,7 +99,7 @@ set(LIBS
     ${Boost_LIBRARIES}
     ${ICU}
     ${RT}
-    ${PROCPS_LDFLAGS})
+)
 target_link_libraries(${CMAKE_PROJECT_NAME} ${LIBS})
 
 webos_build_system_bus_files()
